--- a/source/world/gunfire.cpp
+++ b/source/world/gunfire.cpp
@@ -27,7 +27,7 @@ GunFire::GunFire( Fixed range )
     m_distanceToTarget(0)
 {
     m_range = range;
-    m_speed = Fixed::Hundredths(48);
+    m_speed = Fixed::Hundredths(50);
     m_turnRate = Fixed::Hundredths(80);
     m_maxHistorySize = -1;
     m_movementType = MovementTypeAir;
@@ -168,7 +168,7 @@ bool GunFire::MoveToWaypoint()
     }
 
     if( newDistance <= distToTarget && 
-        newDistance < Fixed::Hundredths(48) &&
+        newDistance < Fixed::Hundredths(50) &&
         m_targetLongitudeAcrossSeam == 0 )
     {
         ClearWaypoints();
@@ -194,7 +194,7 @@ void GunFire::CalculateNewPosition( Fixed *newLongitude, Fixed *newLatitude, Fi
     
     Fixed distance = (Vector3<Fixed>( m_targetLongitude, m_targetLatitude, 0 ) -
                       Vector3<Fixed>( m_longitude, m_latitude, 0 )).Mag();
-    m_speed = distance / 48;
+    m_speed = distance / 50;
 
     Fixed minSpeed = Fixed::Hundredths(40) / g_app->GetWorld()->GetGameScale();
     Fixed maxSpeed = 2 / g_app->GetWorld()->GetGameScale();
--- a/source/world/sub.cpp
+++ b/source/world/sub.cpp
@@ -167,7 +167,7 @@ bool Sub::Update()
             m_targetObjectId = -1;
             if( g_app->GetWorld()->GetDefcon() < 4 )
             {
-                m_retargetTimer = 4 + syncfrand(11);
+                m_retargetTimer = 5 + syncfrand(10);
                 WorldObject *obj = FindTarget();
                 if( obj )
                 {
                    m_targetObjectId = obj->m_objectId;
                }                
            }
--- a/source/world/world.cpp
+++ b/source/world/world.cpp
@@ -1815,10 +1815,6 @@ void World::Update()
             m_teams[i]->m_sharingRadar.SetAll( true );
         }
     }
-    
-    START_PROFILE( "Radar Coverage" );
-    UpdateRadar();
-    END_PROFILE( "Radar Coverage" );
 
 
     //
@@ -1970,6 +1966,11 @@ void World::Update()
     END_PROFILE( "GunFire" );
 
     
+    START_PROFILE( "Radar Coverage" );
+    UpdateRadar();
+    END_PROFILE( "Radar Coverage" );
+
+    
     m_votingSystem.Update();
 
     //
